libdnf/rpm/repo.cpp

```diff
@@ -86,6 +90,26 @@ const int COUNTME_VERSION = 0;
 // where each letter represents a window step (starting from 0)
 const std::array<const int, 3> COUNTME_BUCKETS = {{2, 5, 25}};
 
+
+namespace {
+
+/// Throw libdnf::RuntimeError when problem detected
+void is_readable_rpm(const char * fn) {
+    if (access(fn, R_OK) != 0) {
+        const char * err_txt = strerror(errno);
+        throw libdnf::RuntimeError(fmt::format(_("Failed to access RPM: \"{}\": {}"), fn, err_txt));
+    }
+
+    auto len = strlen(fn);
+
+    if (len <= 4 || (strcmp(fn + len - 4, ".rpm") != 0)) {
+        throw libdnf::RuntimeError(fmt::format(_("Failed to read RPM: \"{}\": {}"), fn, "does't have extension \".rpm\""));
```

Comment:
1. The line is too long. Please, try using clang-format.
2. Anonymous namespaces can be nested. Code will be in right namespace and length of line will be OK.
```
namespace libdnf {
namespace {
 // The code is defined in anonymous namespace under libdnf namespace
 // and has access to libdnf namespace by default.
 // So, throw RuntimeError("something"); without libdnf:: namespace prefix can be used.
}
}
```